gems/pending/appliance_console/logfile_configuration.rb

```diff
@@ -8,38 +8,91 @@ module ApplianceConsole
+                  :confirmed_new_logrotate_count, :requested_new_logrotate_count, :activated_new_logrotate_count
 
     include ApplianceConsole::Logging
 
+    def initialize
+      self.requested_new_disk            = false
+      self.confirmed_new_disk            = false
+      self.activated_new_disk            = false
+      self.requested_new_logrotate_count = false
+      self.confirmed_new_logrotate_count = false
+      self.activated_new_logrotate_count = false
+
+      params = { "-h" => nil, "--output=size" => nil, nil => LOGFILE_DIRECTORY }
+
+      self.size = AwesomeSpawn.run!("df", :params => params).output.split("\n")[1].strip
+      self.logrotate_count = File.readlines(MIQ_LOGS_CONF).detect { |line| line =~ /rotate\s*\d.*$/ }.split(" ")[1].to_i
+    end
+
     def activate
-      stop_evm
-      initialize_logfile_disk
-      start_evm
+      activated_new_disk = activate_new_disk if confirmed_new_disk
+      activated_new_logrotate_count = activate_new_logrotate_count if confirmed_new_logrotate_count
+      activated_new_disk || activated_new_logrotate_count
     end
 
     def ask_questions
       clear_screen
-      return false unless use_new_disk
-      choose_disk
+      choose_disk if use_new_disk
+      choose_logrotate_count if set_new_logrotate_count
       confirm_selection
     end
 
     private
 
     def confirm_selection
-      agree("Continue with disk: #{disk.path}: #{disk.size.to_i / 1.megabyte} MB (Y/N):")
+      if requested_new_disk
+        self.confirmed_new_disk =
+          agree("Confirm continue with new logfile disk: #{disk.path}: #{disk.size.to_i / 1.megabyte} MB (Y/N):")
+      end
+
+      if requested_new_logrotate_count
+        self.confirmed_new_logrotate_count =
+          agree("Confirm update the save logrotate count to: #{logrotate_count} (Y/N):")
+      end
+
+      confirmed_new_disk || confirmed_new_logrotate_count
     end
 
     def use_new_disk
-      agree("Configure a new logfile disk volume? (Y/N):")
+      self.requested_new_disk = agree("Configure a new logfile disk volume? (Y/N):")
     end
 
     def choose_disk
       self.disk = ask_for_disk("logfile disk")
     end
 
+    def set_new_logrotate_count
+      self.requested_new_logrotate_count = agree("Change the saved logrotate count from #{logrotate_count}? (Y/N):")
+    end
+
+    def choose_logrotate_count
+      say "\t1G of disk is recommended for each saved logrotate."
+      say "\tThe proposed new disk is #{disk.size.to_i / 1.megabyte} MB" if requested_new_disk
+      say "\tThe current log disk is #{size}" unless requested_new_disk
+
+      self.logrotate_count = ask_for_integer("new log rotate count")
+    end
+
+    def activate_new_logrotate_count
+      say 'Activating new logrotate count'
+      updated = File.readlines(MIQ_LOGS_CONF).each { |line| line.gsub!(/rotate\s*\d.*$/, "rotate #{logrotate_count}") }
+      File.open(MIQ_LOGS_CONF, "w") { |f| updated.each { |line| f.puts line } }
```

Comment:
I think this would be easier to read if you just read in the whole file as a string, then did your `gsub!` on that and rewrote it. That would get rid of all the blocks here.

Maybe something like [this](https://github.com/ManageIQ/linux_admin/blob/master/lib/linux_admin/chrony.rb#L11-L13)?